.github/workflows/e2e.yml

```diff
@@ -10,35 +10,47 @@ on:
 jobs:
   e2e:
     timeout-minutes: 15
-    runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
+        os: [ubuntu-latest, macos-latest]
         head: [headful, headless]
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - name: Install chrome-dev
-        run: |
-          wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | sudo apt-key add -
-          sudo sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list'
-          sudo apt-get update
-          sudo apt-get install -y google-chrome-unstable
       - name: Set up node.js
         uses: actions/setup-node@v3
         with:
           node-version: 16
           cache: npm
       - name: Install and build npm dependencies
         run: npm ci
+      - name: Install Chromium
+        id: chromium
+        run: |
+          # browsers from @puppeteer/browsers
+          version=$(npx browsers install chrome@latest | cut -f 1 -d' ' | sed -e 's/chrome@//')
```

Comment:
with https://github.com/puppeteer/puppeteer/pull/9797 you should be able to do `cut -f 2 -d' '` which would give you the executable path directly.

Reply:
Integrated